extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -410,15 +426,46 @@ private void checkConnection() {
         if (getThing().getStatus() != ThingStatus.ONLINE) {
             openConnection(); // try to reconnect....
         }
-        if (getThing().getStatus() == ThingStatus.ONLINE) {
+
+        if (getThing().getStatus() == ThingStatus.ONLINE && this.session != null && this.session.isOpen()) {
             try {
-                session.getRemote().sendString("HELLO");
-            } catch (IOException | NullPointerException e) {
+                this.session.getRemote().sendPing(null);
+                this.missedPongsCount.countDown();
+            } catch (IOException | NullPointerException | UnsupportedOperationException e) {
+                logger.warn("Sending websocket PING failed.");

```

Comment:
This should be no warning here, only debug.
Furthermore, the message only applies to the `IOException` and `NullPointerException`, while the `UnsupportedOperationException` actually means that we missed some pongs (but actually not for the ping that we just sent, right? So does that make sense here at all?).

Reply:
I'll change the debug and the try/catch behavior here together with the removal of the `Counter`.